client/lib/utils/pdf-support.js

```diff
@@ -0,0 +1,36 @@
+import _ from 'lodash';
+
+/**
+ * This function result is cached for performance reasons, since browser capabilities won't change.
+ *
+ * @returns {string|false} "native" if the browser can display PDFs and code can reliably call JS
+ * functions on those PDFs (such as print()). "addon" if the browser can display PDFs, but there's
+ * no guarantee that they will respond to a DOM-like API. false if the browser can't display
+ * PDFs at all.
+ */
+export default _.memoize( () => {
+	if ( navigator.userAgent.includes( 'Firefox' ) ) {
```

Comment:
`navigator.userAgent` is here since the beginning of JS, and virtually every page in the world relies on it. It may not be considered "best practice" to rely on it, but no browser is going to remove it. AFAIK, there's no other reliable way to check if the browser is `Firefox`.